### PR TITLE
Clean console indexPatterns api error

### PR DESCRIPTION
**Pull request by junqiu-lei**
### Description
Correct indexPatterns api from services.data.indexPatterns to clean browser console error.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by martin-gaievski — APPROVED** [review]

File: public/components/map_container/map_container.tsx [hunk 1]
```diff
@@ -241,8 +241,10 @@ export const MapContainer = ({
       (indexPattern) => indexPattern.id === selectedLayerConfig.source.indexPatternId
     );
     if (!findIndexPattern) {
-      // @ts-ignore
-      const newIndexPattern = await indexPatterns.get(selectedLayerConfig.source.indexPatternId);
+      const newIndexPattern = await services.data.indexPatterns.get(
+        // @ts-ignore
```
Comment (heemin32) [inline public/components/map_container/map_container.tsx:245]: Is this by mistake or intentional? having comment inside method parameter seems weird.

File: public/components/map_container/map_container.tsx [hunk 2]
```diff
@@ -241,8 +241,10 @@ export const MapContainer = ({
       (indexPattern) => indexPattern.id === selectedLayerConfig.source.indexPatternId
     );
     if (!findIndexPattern) {
-      // @ts-ignore
-      const newIndexPattern = await indexPatterns.get(selectedLayerConfig.source.indexPatternId);
+      const newIndexPattern = await services.data.indexPatterns.get(
```
Comment (heemin32) [inline public/components/map_container/map_container.tsx:244]: I just couldn't match commit message and this code change.
`Correct indexPatterns api from services.data.indexPatterns to clean browser console error.`
but the code is changing indexPatterns to services.data.indexPatterns

File: public/components/map_container/map_container.tsx (hunk 1 above)
Comment (junqiu-lei) [reply public/components/map_container/map_container.tsx:245]: Intentional for the below line

File: public/components/map_container/map_container.tsx (hunk 2 above)
Comment (junqiu-lei) [reply public/components/map_container/map_container.tsx:244]: The method is used to update layersIndexPatterns which is only used in top nav comment as a requirement prop, but no use there currently(it's mainly used for global filter, we don't support yet).

It used to be passed in by:
```
    const {
      data: { indexPatterns },
    } = services;
```
so it's the reason we change from `indexPatterns ` to `services.data.indexPatterns`

https://github.com/opensearch-project/dashboards-maps/blob/2.8/public/components/map_top_nav/top_nav_menu.tsx#L141

Comment (heemin32) [reply public/components/map_container/map_container.tsx:244]: Got it. So it is `Use indexPatterns api from services.data.indexPatterns to clean browser console error.`

**Review by heemin32 — APPROVED** [review]

**Pull request merged**